recipes/daggy/2.1/conanfile.py

```diff
@@ -0,0 +1,112 @@
+from conans import ConanFile, CMake, tools
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class DaggyConan(ConanFile):
+    name = "daggy"
+    license = "MIT"
+    homepage = "https://daggy.dev"
+    description = "Data Aggregation Utility and C/C++ developer library for data streams catching"
+    topics = ("streaming", "qt", "monitoring", "process", "stream-processing", "extensible", "serverless-framework", "aggregation", "ssh2", "crossplatform", "ssh-client")
+
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "ssh2_support": [True, False],
+        "yaml_support": [True, False],
+        "console": [True, False],
+        "package_deps": [True, False],
+        "shared": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = {
+        "ssh2_support": True,
+        "yaml_support": True,
+        "console": True,
+        "package_deps": False,
+        "shared": False,
+        "fPIC": False
+    }
+    generators = "cmake", "cmake_find_package"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 20
+
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def build_requirements(self):
+        self.build_requires("cmake/3.22.0")
+
+    def validate(self):
+        if self.settings.get_safe("compiler.cppstd"):
+            tools.check_min_cppstd(self, self._minimum_cpp_standard)
+
+    def configure(self):
+        self.options["qt"].shared = True
```

Comment:
I can not recall, but I think this should be in `config_options` 🤔 let's wait for a second reviewer incase I am mistaken!

When the user's CLI input is read and how it updates the options is a tad confusing for me.

If this is a hard requirement it should be enforce with in the validate method...

Closest existing example I could find https://github.com/conan-io/conan-center-index/blob/f1efcb317eaf0f5257f9165dce062e681b1b8ebd/recipes/hidapi/all/conanfile.py#L34-L38